File: examples/ruby-rspec/steps/roman_numerals.steps.rb
# frozen_string_literal: true

require 'varar'
require_relative '../lib/roman_numerals'

steps do
  # Header-bound table: this sensor runs once per row with the row as a hash keyed
  # by header. Returning {"roman" => …} checks that column; "decimal" is an input.
  sensor('a decimal and a roman number') do |_state, row|
    { 'decimal' => row['decimal'], 'roman' => RomanNumerals.to_roman(row['decimal'].to_i) }
  end
end

File: examples/csharp-vstest/steps/roman-numerals.steps.cs
using System.Globalization;
using Varar;
using Varar.Core;
using static Varar.Example.StepHelpers;

namespace Varar.Example;

public static class RomanNumeralsSteps
{
    public static void Register(Steps s)
    {
        // Header-bound: the paragraph names the table's columns, so the step runs once per row,
        // returning the columns it computes; Vár diffs them against the row cells.
        s.Sensor("a decimal and a roman number", (state, row) =>
        {
            var value = AsStr(SMap(row)["decimal"]);
            var roman = RomanNumerals.ToRoman(int.Parse(value, CultureInfo.InvariantCulture));
            return VMap(("decimal", Value.Of(value)), ("roman", Value.Of(roman)));
        });
    }
}
